#! /bin/sh /usr/share/dpatch/dpatch-run
## 86_PATH_MAX.dpatch by  <mbanck@debian.org> and Cyril Brulebois
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch to define PATH_MAX on systems which do not.
## DP: http://bugs.mysql.com/16574 and Debian bug #378949, #348182.

@DPATCH@

--- old/ndb/src/common/logger/FileLogHandler.cpp
+++ new/ndb/src/common/logger/FileLogHandler.cpp
@@ -18,6 +18,10 @@
 #include <FileLogHandler.hpp>
 #include <File.hpp>
 
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
 //
 // PUBLIC
 //
--- old/ndb/src/kernel/blocks/ndbfs/AsyncFile.cpp
+++ new/ndb/src/kernel/blocks/ndbfs/AsyncFile.cpp
@@ -64,6 +64,10 @@
 #define WRITEBUFFERSIZE 262144
 #endif
 
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
 const char *actionName[] = {
   "open",
   "close",
--- old/ndb/src/kernel/blocks/ndbfs/Filename.cpp
+++ new/ndb/src/kernel/blocks/ndbfs/Filename.cpp
@@ -25,6 +25,10 @@
 
 #include <signaldata/FsOpenReq.hpp>
 
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
 static const char* fileExtension[] = {
   ".Data",
   ".FragLog",
--- old/ndb/src/kernel/blocks/ndbfs/Filename.hpp
+++ new/ndb/src/kernel/blocks/ndbfs/Filename.hpp
@@ -17,6 +17,10 @@
 #ifndef Filename_H
 #define Filename_H
 
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
 //===========================================================================
 //
 // .DESCRIPTION
--- old/ndb/src/kernel/vm/Configuration.cpp
+++ new/ndb/src/kernel/vm/Configuration.cpp
@@ -39,6 +39,10 @@
 #include <LogLevel.hpp>
 #include <NdbSleep.h>
 
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
 extern "C" {
   void ndbSetOwnVersion();
 }
--- old/ndb/src/cw/cpcd/CPCD.cpp
+++ new/ndb/src/cw/cpcd/CPCD.cpp
@@ -24,6 +24,10 @@
 
 #include "common.hpp"
 
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
 extern const ParserRow<CPCDAPISession> commands[];
 
 
--- old/ndb/src/cw/cpcd/Process.cpp
+++ new/ndb/src/cw/cpcd/Process.cpp
@@ -27,6 +27,10 @@
 #include <sys/resource.h>
 #endif
 
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
 void
 CPCD::Process::print(FILE * f){
   fprintf(f, "define process\n");
--- old/ndb/tools/restore/Restore.hpp
+++ new/ndb/tools/restore/Restore.hpp
@@ -26,6 +26,10 @@
 #include <ndb_version.h>
 #include <version.h>
 
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
 static const char * delimiter = ";"; // Delimiter in file dump
 
 const int FileNameLenC = 256;
--- old/ndb/include/util/File.hpp
+++ new/ndb/include/util/File.hpp
@@ -19,6 +19,10 @@

 #include <ndb_global.h>

+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
 /**
  * This class provides a file abstraction . It has operations
  * to create, read, write and delete a file.
--- old/ndb/src/common/portlib/NdbConfig.c
+++ new/ndb/src/common/portlib/NdbConfig.c
@@ -20,6 +20,10 @@
 #include <NdbMem.h>
 #include <basestring_vsnprintf.h>

+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
 static const char *datadir_path= 0;

 const char *
